File: 0389-find-the-difference/0389-find-the-difference.py
class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        original = dict()
        extended = dict()
        
        for letter in s:
            original[letter] = original.get(letter, 0) + 1
        
        for letter in t:
            extended[letter] = extended.get(letter, 0) + 1
        
        for letter in extended.keys():
            if letter not in original.keys():
                return letter
            elif original[letter] != extended[letter]:
                return letter

File: 0389-find-the-difference/0389-find-the-difference.swift
class Solution {
    func findTheDifference(_ s: String, _ t: String) -> Character {
        var original = [Character: Int]()
        var extended = [Character: Int]()
        
        for letter in s {
            if original[letter] != nil {
                original[letter]! += 1
            } else {
                original[letter] = 1
            }
        }

        for letter in t {
            if extended[letter] != nil {
                extended[letter]! += 1
            } else {
                extended[letter] = 1
            }
        }
        
        for letter in extended.keys {
            if original[letter] != extended[letter] || original[letter] == nil {
                return letter
            }
        }
        
        return "-"
    }
}